app/assets/javascripts/components/common/ArticleViewer/containers/ArticleViewer.jsx

```diff
@@ -63,9 +63,21 @@ export class ArticleViewer extends React.Component {
   // first in that case. In that case, componentDidUpdate fetches the
   // user ids as soon as usernames are avaialable.
   componentDidUpdate(prevProps, prevState) {
-    if (!prevProps.users && this.props.users) {
-      if (!prevState.userIdsFetched) {
-        this.fetchUserIds(this.props.users);
+    // checks for the availability of 'assignedUsers' and 'users' props.
```

Comment:
These comments are helpful, but I think they might point the way to some further simplification that is possible.

Here's my understanding:

Wherever `ArticleViewer` is used, the `users` prop represents all the relevant users who have edited the article, and it won't be passed initially...it is only passed once fetchArticleDetails returns data for which users in the course have edited the article. So any time `prevProps.users` is empty but `props.users` is not empty, it means it's time to fetch user Ids if they haven't been fetched yet.

This logic holds whether `assignedUsers` is passed or not, so we could get rid of the `else` branch altogether by using something that works whether or not `assignedUsers` is there, like `union(this.props.assignedUsers || [] , this.props.users)`

Reply:
Thanks alot for the input, i have improved upon the solution by keeping all these things in mind, just waiting for travis-ci to finish.

Reply:
Hello @ragesoss, i have updated the solution, please review it.